contracts/PlasmaCash.sol

```diff
@@ -181,19 +181,21 @@ contract PlasmaCash is Withdrawable, Operable {
   }
 
 
-  function validateProofSignaturesAndTxData(bytes _txBytes, bytes _proof, bytes _signature, address _signer, uint256 _targetBlock)
-  public
+  function validateProofSignaturesAndTxData(Transaction.TX memory transaction, bytes _proof, bytes _signature, address _signer, uint256 _targetBlock, bool checkSig)
+  private
```

Comment:
with public, we allow any user to validate on-chain... any reason why you want this to be private?